File: 3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        flip = 0

        for i in range(n - 2):
            if nums[i] == 0:
                nums[i] ^= 1
                nums[i + 1] ^= 1
                nums[i + 2] ^= 1
                flip += 1

        for i in range(n - 2, n):
            if nums[i] == 0:
                return -1

        return flip

File: LeetCode/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        int flip = 0;

        for (int i = 0; i <= n - 3; ++i) {
            if (nums[i] == 0) {
                nums[i] ^= 1;
                nums[i+1] ^= 1;
                nums[i+2] ^= 1;
                flip++;
            }
        }

        for (int i = n - 2; i < n; ++i) {
            if (nums[i] == 0) {
                return -1;
            }
        }

        return flip;
    }
};